openvdb_houdini/SOP_NodeVDB.cc

```diff
@@ -353,107 +352,6 @@ SOP_NodeVDB::getNodeSpecificInfoText(OP_Context &context, OP_NodeInfoParms &parm
-    // traverse upstream nodes, if unload is not possible, duplicate the source
-    if (!isSourceStealable(index, context)) {
-        duplicateSource(index, context, *pgdp, clean);
-        unlockInput(index);
-        return error();
-    }
-    OPENVDB_NO_DEPRECATION_WARNING_END
-
-    // get the input GU_Detail handle and unlock the inputs
-    GU_DetailHandle inputgdh = inputGeoHandle(index);
-
-    unlockInput(index);
-    SOP_Node *input = CAST_SOPNODE(getInput(index));
-
-    if (!input) {
-        addError(SOP_MESSAGE, "Invalid input SOP Node when attempting to unload.");
-        return error();
-    }
-
-    // explicitly unload the data from the input SOP
-    const bool unloadSuccessful = input->unloadData();
-
-    // check if we only have one reference
-    const bool soleReference = (inputgdh.getRefCount() == 1);
-
-    // if the unload was unsuccessful or the reference count is not one, we fall back to
-    // explicitly copying the input onto the gdp
-    if (!(unloadSuccessful && soleReference)) {
-        const GU_Detail *src = inputgdh.readLock();
-        UT_ASSERT(src);
-        if (src)  (*pgdp)->copy(*src);
-        inputgdh.unlock(src);
-        return error();
-    }
-
-    // release our old write lock on gdp (setup by cookMe())
-    gdh.unlock(*pgdp);
-    // point to the input's old gdp and setup a write lock
-    gdh = inputgdh;
-    *pgdp = gdh.writeLock();
-
-    return error();
-}
-
-
-bool
-SOP_NodeVDB::isSourceStealable(const unsigned index, OP_Context& context) const
-{
-    struct Local {
-        static inline OP_Node* nextStealableInput(
-            const unsigned idx, const fpreal now, const OP_Node* node)
-        {
-            OP_Node* input = node->getInput(idx);
-            while (input) {
-                OP_Node* passThrough = input->getPassThroughNode(now);
-                if (!passThrough) break;
-                input = passThrough;
-            }
-            return input;
-        }
-    }; // struct Local
-
-    const fpreal now = context.getTime();
-
-    for (OP_Node* node = Local::nextStealableInput(index, now, this); node != nullptr;
-        node = Local::nextStealableInput(index, now, node))
-    {
-        // cont'd if it is a SOP_NULL.
-        std::string opname = node->getName().toStdString().substr(0, 4);
-        if (opname == "null") continue;
-
-        // if the SOP is a cache SOP we don't want to try and alter its data without a deep copy
-        if (dynamic_cast<SOP_Cache*>(node))  return false;
-
-        if (node->getUnload() != 0)
-            return true;
-        else
-            return false;
-    }
-    return false;
-}
-
-
-OP_ERROR
-SOP_NodeVDB::duplicateSourceStealable(const unsigned index, OP_Context& context)
-{
-    OPENVDB_NO_DEPRECATION_WARNING_BEGIN
-    auto error = this->duplicateSourceStealable(index, context, &gdp, myGdpHandle, true);
-    OPENVDB_NO_DEPRECATION_WARNING_END
-    return error;
-}
-
-
-////////////////////////////////////////
```

Comment:
So we still use these heavily. I would propose that these aren't removed until we have full DS support. If we really want to get rid of them I can re-introduce them on my end so it's not a huge problem, but it's worth pointing out that they're still in use somewhere

Reply:
Is there any specific features of DS that you're after that's holding you up in migrating? As much as possible, I think we should avoid maintaining code that's only there for one studio to use. As you say it's easy enough for that studio to maintain it in their own fork of the code so that should be preferred in these types of cases. Happy to discuss in our next TSC meeting though.

Reply:
Two parts to this. The first is that, when I last profiled, this method was still slightly faster than the compiled blocks, primarily for small-ish VDB geo streams, due to what I can only assume is the way some parameters were being evaluated. The second and more important issue is that we derive from this SOP type extensively throughout a good number of nodes which rely on this method. Along with the standard VDB Houdini nodes, these have subsequently been used in many digital assets (a good 50+ at least) and in general networks too. They rely on this unload behaviour for performance as there's no way we've been able to go through every asset and upgrade to compiled blocks. 

I don't believe this to be studio specific; anyone that previously relied on this behaviour will suddenly notice a change in performance. Thinking about it, this deprecation warning is actually a bit pointless. Really we could have done with something like a SOP warning when `duplicateSourceSteable` was being used in an unload context. Lets definitely discuss in TSC!

Reply:
I ran a very quick test with some of our networks by switching to the C++ verb support (our software already supports it, it's just turned off). As expected, a bunch of HDA's started evaluating every timestep when they didn't need to - however more concerning was that fact that some of the SOPs in question, primarily smaller operators, increased in cook time by a factor of x2/3 according to the performance monitor. Our internal operator timers come out with the same time execution so perhaps this is expected? i.e. maybe the performance monitor isn't actually working correctly with duplicate source stealable? @jmlait any thoughts?

Reply:
Marshalling through the UT_Options the way the NodeParmsOptions does would be slower than directly accessing evalString().  But I'm a bit surprised at 2/3x!

On the other hand, our perf monitor is known to have strange things, so maybe it is miscounting?  Maybe wall clock the time to cook the chain?